src/App.tsx

```diff
@@ -15,36 +16,94 @@
   'Garlic',
 ];
 
+enum SortBy {
+  None,
+  Abc,
+  Length,
+}
+
+const sortGoods = (sortField: SortBy, reverseField: boolean) => {
+  const sortedGoods = [...goodsFromServer];
+
+  switch (sortField) {
+    case SortBy.Abc:
+      sortedGoods.sort((a, b) => a.localeCompare(b));
+      break;
+
+    case SortBy.Length:
+      sortedGoods.sort((a, b) => a.length - b.length);
+      break;
+
+    default:
+  }
+
+  if (reverseField) {
+    sortedGoods.reverse();
+  }
+
+  return sortedGoods;
+};
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = React.useState(SortBy.None);
+  const [reverseField, setReverseField] = React.useState(false);
+  const goods: string[] = sortGoods(sortField, reverseField);
+
+  const reset = () => {
+    setSortField(SortBy.None);
+    setReverseField(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortField !== SortBy.Abc,
+          })}
+          onClick={() => setSortField(SortBy.Abc)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={cn('button is-success', {
+            'is-light': sortField !== SortBy.Length,
+          })}
+          onClick={() => setSortField(SortBy.Length)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={cn('button is-warning', {
+            'is-light': reverseField !== true,
+          })}
+          onClick={() => setReverseField(!reverseField && true)}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {goods.toString() !== goodsFromServer.toString() && (
```

Comment:
The condition for displaying the `Reset` button checks if the `goods` array is different from `goodsFromServer`. This can be optimized by checking if `sortField` is not `SortBy.None` or `reverseField` is true, which directly indicates that sorting or reversing has been applied.